test/index.spec.js

```diff
@@ -0,0 +1,265 @@
+          app_id: 'com.test.app.id',
+          from:   '618984568766',
+        },
+      };
+      let run = initFetchLatest();
+      await run(utils.event, utils.res, utils.index);
+      utils.res.status.should.have.been.calledWith(200);
+    });
+
+    it('should return the data', async() => {
+      let utils = initGetTest();
+      utils.event = {
+        url: `/fetch_latest?app_id=com.test.app&from=618984568766`,
+        query: {
+          app_id: 'com.test.app.id',
+          from:   '618984568766',
+        },
+      };
+      let run = initFetchLatest();
+      await run(utils.event, utils.res, utils.index);
+      utils.res.json.should.have.been.calledWith(utils.data);
+    });
+
+    it('should return a 400 response on malformed app id', async () => {
+      let utils = initGetTest();
+      utils.event = {
+        url: `/fetch_latest?app_id=com.test.@pp&from=618984568766`,
+        query: {
+          app_id: 'com.test.@pp',
+          from: '618984568766',
+        },
+      };
+      let run = initFetchLatest();
+      await run(utils.event, utils.res, utils.index);
+      utils.res.status.should.have.been.calledWith(400);
+    });
+    it('should return a 400 response on malformed cursor', async () => {
+      let utils = initGetTest();
+      utils.event = {
+        url: `/fetch_latest?app_id=com.test.app&from=cursor`,
+        query: {
+          app_id: 'com.test.app.id',
+          from: 'cursor',
+        },
+      };
+      let run = initFetchLatest();
+      await run(utils.event, utils.res, utils.index);
+      utils.res.status.should.have.been.calledWith(400);
+    });
+
+    it('should return a 500 response if fetch fails', async () => {
+      let utils = initGetTest();
+      utils.queryStub.createQueryJob.rejects();
+      utils.event = {
+        url: `/fetch_latest?app_id=com.test.app&from=618984568766`,
+        query: {
+          app_id: 'com.test.app.id',
+          from: '618984568766',
+        },
+      };
+      let run = initFetchLatest();
+      await run(utils.event, utils.res, utils.index);
+      utils.res.status.should.have.been.calledWith(500);
+    });
+
+    it('should throw an error if auth fails for the table', async () => {
+      let utils = initGetTest();
+      utils.event = {
+        url: `?app_id=com.test.app&from=654982341687`,
+        query: {
+          app_id: 'com.test.app',
+          from: '654982341687',
+        },
+      };
+      const err = new Error('you do not have permission to access this table');
+      utils.queryStub.createQueryJob.throws(err);
+      const spy = sandbox.spy(utils.index, 'fetchLatestHandler');
+      let run = initFetchLatest();
+      try {
+        await run(utils.event, utils.res, utils.index);
+        utils.res.status.should.have.been.calledWith(500);
+      } catch (e) {
+        utils.index.server.close();
+        console.log(e);
+        e.message.should.equal(err.message);
+      }
+    });
+  });
+  describe('setTable', () => {
```

Comment:
`getTable` is tested both explicitly and implicitly in these tests. That's fine, but it should be trivial to test explicitly, it only has one dependency (the table json file). Why not have it take that dependency explicitly: 

`function getTable(table, id)`

Then you can test it without stubbing anything, you just give it the table and the id and show it throws the right errors. 

The handler then becomes dependent on the table file, but it was indirectly anyway, so no change there!